src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1125,14 +1132,19 @@ def random_mac(self):
     @item_method
     @accepts(Int('id'), Dict('options', Bool('overcommit')))
     async def start(self, id, options):
-        """Start a VM.
-
+        """
+        Start a VM.
         options.overcommit defaults to false, which means VM will not be allowed to
         start if there is not enough available memory to hold all VMs configured memory.
-        If true VM will start even if there is not enough memory for all VMs configured memory."""
-
+        If true VM will start even if there is not enough memory for all VMs configured memory.
+        """
         vm = await self._get_instance(id)
 
+        verrors = ValidationErrors()
+        await self.__common_validation(verrors, 'vm_start', vm)
```

Comment:
It seems overkill to validatye all data again on vm.start? Perhaps we should just call vm.flags here and raise a CallError if not supported? Thoughts?

Reply:
Running through all of validation again ensures that everything is still correct, there could be a significant period of time between creation and start, including a bad database perhaps. This avoids a nasty set of issues later on, and doesn't take too much time

Reply:
Except we are now using to validate code on a place not prepared to handle the exception we are now raising and making the validation code dirty with a CallError and schema name